examples/run_config.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+"""
+Using :class:`~caf.toolkit.BaseConfig`
+======================================
+
+The :class:`~caf.toolkit.BaseConfig` class in caf.toolkit is designed to load
+and validated YAML configuration files. This example shows how
+to create child classes of BaseConfig to load parameters.
+
+See Also
+--------
+:class:`~caf.toolkit.arguments.ModelArguments`
+    for producing command-line arguments from a BaseConfig class.
+"""
+
+# Built-Ins
+import pathlib
+import re
+from typing import Any, Self
+
+# Third Party
+import pydantic
+from pydantic import dataclasses
+
+# Local Imports
+import caf.toolkit as ctk
+
+folder = pathlib.Path().parent
+
+# %%
+# Basic
+# -----
+#
+# This example shows how to create a simple configuration file, without nesting, and
+# with different types of parameters. The majority of Python built-in types can be used,
+# additionally dataclasses and many "simple"[#simple]_ custom types can be used.
+#
+# .. seealso ::
+#       :ref:`Extra Validation` for information on more complex validation.
+#
+# The :class:`pydantic.DirectoryPath` and :class:`pydantic.FilePath` types both return
+# :class:`pathlib.Path` objects after validating that the directory, or file, exists.
```

Comment:
I think this would be better is a "note" box, similar to the "see also" above. Comment below has more detail